### DISCUSSION TITLE
HW 1 Part 2 #1

### DISCUSSION THREAD

**Discussion by courtcarroll** [cooperrc/computational-mechanics#86] [Q&A]
For problem one I am receiving an error in my code that the array is not defined. I have included a picture of it I'm not sure if I have an error in my code or if I am completely off with my code any advice is welcome!

<img width="1109" alt="Screen Shot 2022-09-08 at 10 35 11 PM" src="https://user-images.githubusercontent.com/112587455/189260905-b935bf63-ceb9-4c3a-ad17-8bb35f6ae46d.png">

Also if my code has a def function and a return function but there is no error or output printed does that mean the code is incorrect?

**Comment by chrisbergen4** [comment]
def sincos(c):    
    
    '''help'''
    
    sinx = np.sin(c)
    cosx = np.cos(c)
    return sinx, cosx

z = sincos(x)
sinx = z[0]
cosx = z[1]

**Reply by chrisbergen4** [reply]
I was having the same issue. I solved it basically by having the function return an array "z", where the z[0] was the array of sinx, and z[1] was the array of cosx. There might be a way to have the returned variables exist outside the function, but it was taking too long, so i just did that instead and it worked

**Comment by cooperrc** [comment]
This is a strange concept at first, but it can be really powerful once you get the hang of it. 

When you "run" the `def sincos(x)` to create a function, it creates a new variable called `sincos` that _is_ a function. **But** it does not run the function until you ask it to e.g. `sincos(0)` returns `(0, 1)` _sin(0) and cos(0)_.
